doc: add recommendations about resource limits optimizations (#5276)
operator-framework/operator-sdk: website/content/en/docs/best-practices/common-recommendation.md

    - For Ansible-based operators, you can also use [Molecule][molecule], an Ansible testing framework. For further information see [Testing with Molecule][molecule-tests]
    - For Helm-based operators, you can also use [Chart tests][helm-chart-tests]
- Ensure that you checked the [Can I customize the projects initialized with operator-sdk?][faq] and understand the [Project Layout][project-layout] before starting to do your customizations as please you on top.

> - Optimize the values of the manager resources in `config/manager/manager.yaml` according to the project requirements. It is recommend defined the resources limits in order to follow good practices and for security reasons. More info: [Managing Resources for Containers][k8s-manage-resources] and [Docker Security Cheat Sheet][docker-cheats].

Reviewer: ```suggestion
- Optimize manager resource values in `config/manager/manager.yaml` according to project requirements. It is recommended to define resources limits in order to follow good practices and for security reasons. More info: [Managing Resources for Containers][k8s-manage-resources] and [Docker Security Cheat Sheet][docker-cheats].
```

> - Look for `TODO(user)` into the source code generated by the CLI to ensure that you will able to do the suggested customizations.

Reviewer: ```suggestion
- Look for `TODO(user)` in the source code generated by the CLI to ensure that you follow all suggested customizations.
```
